src/structs/mod.rs

```diff
@@ -1 +1,3 @@
+pub mod archival_network_response;
+pub mod error;
 pub mod internet_archive_urls;
```

Comment:
I think it would be better to keep related structs under the related logic. Instead of having one place storing all the structs of various unrelated parts of the programs, I would put them under the module they are actually used in.

`ArchivalResponse` can be in `crate::archival::archival_response`, or even  `crate::archival::models::archival_response`. `ArchivalError` can be in `crate::archival::error`, etc...

Reply:
Yupp, restructured ✅